tools/ci_build/github/azure-pipelines/linux-ci-pipeline.yml

```diff
@@ -108,13 +108,47 @@ stages:
                         --test;"
                 workingDirectory: $(Build.SourcesDirectory)
 
+      - bash: |
+          mkdir -p $HOME/.onnx
+          docker run --rm \
+            --volume /data/onnx:/data/onnx:ro \
+            --volume $(Build.SourcesDirectory):/onnxruntime_src \
+            --volume $(Build.BinariesDirectory):/build \
+            -e ALLOW_RELEASED_ONNX_OPSET_ONLY=0 \
+            -e NIGHTLY_BUILD \
+            -e BUILD_BUILDNUMBER \
+            onnxruntimecpubuildcentos8x64 \
+                /bin/bash /onnxruntime_src/tools/scripts/python_test.sh /onnxruntime_src /build Debug
+        displayName: 'Run Debug tests'
+
+      - bash: |
+          mkdir -p $HOME/.onnx
+          docker run --rm \
+            --volume /data/onnx:/data/onnx:ro \
+            --volume $(Build.SourcesDirectory):/onnxruntime_src \
+            --volume $(Build.BinariesDirectory):/build \
+            -e ALLOW_RELEASED_ONNX_OPSET_ONLY=0 \
+            -e NIGHTLY_BUILD \
+            -e BUILD_BUILDNUMBER \
+            onnxruntimecpubuildcentos8x64 \
+            /bin/bash -c "
+              set -ex; \
+              pushd /onnxruntime_src/csharp; \
+              dotnet restore /onnxruntime_src/csharp/OnnxRuntime.DesktopOnly.CSharp.sln; \
+              dotnet build /onnxruntime_src/csharp/OnnxRuntime.DesktopOnly.CSharp.sln; \
+              dotnet test /onnxruntime_src/csharp/OnnxRuntime.DesktopOnly.CSharp.sln -f net6.0 --no-build -l \"console;verbosity=normal\"; \
```

Comment:
Were you able to make it work? I tried to add "--build_nuget" to build.py's flags, then I noticed that in build.py we explicitly disabled the C# test for non-Windows platforms. And, these tests depends on .Net's ios workload which is not available on Linux.

Reply:
OnnxRuntime.DesktopOnly.CSharp.sln excludes all the mobile workloads.

Reply:
The run_csharp_tests function in build.py uses Microsoft.ML.OnnxRuntime.Tests.NetCoreApp.csproj . What should it be replaced with?

Reply:
NetCoreApp is for all platforms. Is that not working somewhere?

There are separate test projects for iOS and Android

Reply:
NetCoreApp requires ios workload to be installed, but that workload is not available on Linux. So we cannot run the test (the run_csharp_tests  function in build.py) on Linux.

Reply:
Here is the log:
```
2024-02-24T20:15:19.3548279Z 2024-02-24 20:15:19,354 build [INFO] - dotnet test /onnxruntime_src/csharp/test/Microsoft.ML.OnnxRuntime.Tests.NetCoreApp/Microsoft.ML.OnnxRuntime.Tests.NetCoreApp.csproj --filter FullyQualifiedName!=Microsoft.ML.OnnxRuntime.Tests.InferenceTest.TestPreTrainedModels  /p:OnnxRuntimeBuildDirectory="/build"
2024-02-24T20:15:19.9352027Z   Determining projects to restore...
2024-02-24T20:15:20.3906730Z /usr/share/dotnet/sdk/8.0.200/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.EolTargetFrameworks.targets(38,5): error NETSDK1202: The workload 'android' is out of support and will not receive security updates in the future. Please refer to https://aka.ms/maui-support-policy for more information about the support policy. [/onnxruntime_src/csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj::TargetFramework=net6.0-android]
2024-02-24T20:15:20.3997572Z /usr/share/dotnet/sdk/8.0.200/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets(117,5): error NETSDK1139: The target platform identifier ios was not recognized. [/onnxruntime_src/csharp/src/Microsoft.ML.OnnxRuntime/Microsoft.ML.OnnxRuntime.csproj::TargetFramework=net6.0-ios]
```
https://dev.azure.com/onnxruntime/2a773b67-e88b-4c7f-9fc0-87d31fea8ef2/_apis/build/builds/1303499/logs/72

Reply:
The `dotnet test` command needs to specify the current value for IncludeMobileTargets as a property (`/p:IncludeMobileTargets=<value>`)

`dotnet test /onnxruntime_src/csharp/test/Microsoft.ML.OnnxRuntime.Tests.NetCoreApp/Microsoft.ML.OnnxRuntime.Tests.NetCoreApp.csproj --filter FullyQualifiedName!=Microsoft.ML.OnnxRuntime.Tests.InferenceTest.TestPreTrainedModels  /p:OnnxRuntimeBuildDirectory="/build"`

like we do when running other commands

`dotnet restore OnnxRuntime.DesktopOnly.CSharp.sln --configfile NuGet.CSharp.config /p:IncludeMobileTargets=false`

Reply:
We usually do that implicitly in build.py but it is setup so CIs can easily provide the value as well.

https://github.com/microsoft/onnxruntime/blob/8a71b657654d63437267014b324bf124a80de347/tools/ci_build/build.py#L2311-L2313